.github/workflows/ci.yaml

```diff
@@ -5,28 +5,34 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10']
+        versions:
+          # test earliest biopython version supporting each python version
+          # from https://github.com/biopython/biopython/blob/master/NEWS.rst
+          - {python: '3.7', biopython: '1.73'}
+          - {python: '3.8', biopython: '1.76'}
+          - {python: '3.9', biopython: '1.79'}
+          - {python: '3.10', biopython: ''} # empty string = latest supported
```

Comment:
While testing the earliest version is useful, the ideal is still an n×m test matrix like you started with that crosses several BioPython versions with several Python versions. This was my intent in the [comment motivating this change](https://github.com/nextstrain/augur/issues/798#issuecomment-993044873), which notes:

> [Always testing] earliest [instead of latest] bakes in the assumption that we're more likely to find more bugs in the older versions than the newer versions, but I'm not convinced by that assumption. :-) Compat bugs happen both because we precede the older versions we support but also because we lag newer versions.

So a test matrix like:

```yaml
python-version:
  - '3.7'
  - '3.8'
  - '3.9'
  - '3.10'
biopython-version:
  - '1.73' # first to support Python 3.7
  - '1.76' # first to support Python 3.8
  - '1.79' # first to support Python 3.9
  - '1.80' # first to support Python 3.10
  - ''     # latest
```

where unsupported combos are then excluded/skipped. One way to do this is to use [exclusion rules](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-configurations-from-a-matrix) in the matrix definition. Another is to use a `job.test.if` statement that skips based on the [`matrix` context](https://docs.github.com/en/actions/learn-github-actions/contexts#matrix-context).

Reply:
@tsibley I switched back to the combination-style matrix with the `exclude` option: https://github.com/nextstrain/augur/pull/843/commits/2ef5fbd791e37000ff5e52a2b887d2207ad00699

Reply:
Thanks! Would you keep 1.80? I included it in my example so that we keep testing it when subsequent versions of BioPython are released.

Reply:
1.80 isn't released yet, so it wouldn't work:

```
% mamba create -n test biopython=1.80
...
Encountered problems while solving:
  - nothing provides requested biopython 1.80**
```

Or do you mean add it for now but exclude from running on all Python versions until it is released?

Reply:
Oh. Sorry! I somehow missed that entirely when I had been looking at BioPython's changelog earlier. :man_facepalming: 

![image](https://user-images.githubusercontent.com/79913/152453302-b9206137-14bf-4425-ac16-74389677fb02.png)

orz orz orz

Reply:
Haha, easy to skim over the large text when it takes some effort to parse the small text for individual Python versions.